### ISSUE TITLE
aws_cloudwatch_event_rule - error reading CloudWatch Events Rule - unexpected format for ID

### ISSUE THREAD

**Issue by sufiyanghori-apt** [hashicorp/terraform-provider-aws#20605]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.4
provider registry.terraform.io/hashicorp/aws v3.54.0

### Affected Resource(s)

* aws_cloudwatch_event_rule

### Terraform Configuration Files

```hcl
resource "aws_cloudwatch_event_rule" "buildkite-secret-scanner-events" {
  name          = "test"
  event_bus_name = "aws.partner/foo.com/foo/18554d09-58ff-aa42-ba9c-c4c33899006f"
  event_pattern = <<PATTERN
{
  "detail-type": [ "test" ]
}
PATTERN
}
```

### Expected Behavior

Event rule should be created.

### Actual Behavior

Terraform fails,

https://github.com/hashicorp/terraform-provider-aws/issues/16973#issuecomment-817020861

```
error reading CloudWatch Events Rule (aws.partner/appflow/salesforce.com/xxx/foo-eventbus/bar-rule): unexpected format for ID ("aws.partner/appflow/salesforce.com/xxx/foo-eventbus/bar-rule"), expected / or "
```

### Steps to Reproduce

1. `terraform apply`

### References

https://github.com/hashicorp/terraform-provider-aws/issues/16973#issuecomment-817020861

* #0000

**Comment by ewbankkit** [comment]
Related:

* #8759
* #19234

**Comment by ewbankkit** [comment]
@sufiyanghori-apt Thanks for raising this issue.
This should have been fixed in #19072, released in **v3.39.0** of the Terraform AWS Provider.
Does your event bus name contain any "special" characters?
